libsql-server/tests/standalone/attach.rs

```diff
@@ -214,3 +215,96 @@ fn attach_auth() {
 
     sim.run().unwrap();
 }
+
+#[test]
+fn attach_auth_with_uuids() {
+    let mut sim = turmoil::Builder::new().build();
+
+    sim.host("primary", make_standalone_server);
+
+    sim.client("test", async {
+        let client = Client::new();
+
+        let (enc, jwt_key) = key_pair();
+
+        let main_db_id = Uuid::new_v4();
+        let attach_db_id = Uuid::new_v4();
+
+        assert!(client
+            .post(
+                format!("http://primary:9090/v1/namespaces/{}/create", main_db_id).as_str(),
+                serde_json::json!({ "jwt_key": jwt_key })
+            )
+            .await
+            .unwrap()
+            .status()
+            .is_success());
+        assert!(client
+            .post(
+                format!("http://primary:9090/v1/namespaces/{}/create", attach_db_id).as_str(),
+                serde_json::json!({ "allow_attach": true, "jwt_key": jwt_key })
+            )
+            .await
+            .unwrap()
+            .status()
+            .is_success());
+
+        let claims = serde_json::json!({
+            "p": {
+                "rw": {
+                    "ns": [main_db_id, attach_db_id]
+                },
+                "roa": {
+                    "ns": [attach_db_id]
+                }
+            }
+        });
+        let token = encode(&claims, &enc);
+
+        let attach_conn = Database::open_remote_with_connector(
+            format!("http://{}.primary:8080", attach_db_id).as_str(),
+            &token,
+            TurmoilConnector,
+        )?
+        .connect()
+        .unwrap();
+        attach_conn
+            .execute("CREATE TABLE bar_table (x)", ())
+            .await
+            .unwrap();
+        attach_conn
+            .execute("insert into bar_table values (43)", ())
+            .await
+            .unwrap();
+
+        let main_conn = Database::open_remote_with_connector(
+            format!("http://{}.primary:8080", main_db_id).as_str(),
+            &token,
+            TurmoilConnector,
+        )?
+        .connect()
+        .unwrap();
+
+        // fails: namespace is uuid, hence needs to be wrapped in quotes
+        assert_debug_snapshot!(main_conn
+            .execute(
+                "ATTACH DATABASE ae308915-caca-480f-a6b4-9f9f9dc84b11 as bar",
```

Comment:
I included this too, since I wanted to check the parser as well. I have hardcoded the UUID (which doesn't matter) since I am using snapshot tests.